src/Symfony/Component/Security/Http/EventListener/LoginThrottlingListener.php

```diff
@@ -30,7 +29,7 @@ final class LoginThrottlingListener implements EventSubscriberInterface
     private $requestStack;
     private $limiter;
 
-    public function __construct(RequestStack $requestStack, Limiter $limiter)
+    public function __construct(RequestStack $requestStack, AbstractRequestRateLimiter $limiter)
```

Comment:
I think it would be better to typehint an interface than an abstract class. This way, it makes it easier for cases where composition is necessary (as you don't need to extend the abstract class without using its implementation)

Reply:
Hmm, alright. Added the interface + fixed this typehint :)